tools/scripts/gitrelease.pl

```diff
@@ -1799,95 +1955,120 @@ sub message_news_template_file_section {
 
 sub remedy_news_template_file_section {
   my $settings = shift();
-
-  insert_news_template("X.Y.Z", $timestamp_marker);
-
+  insert_news_template($settings, 1);
   return 1;
 }
 
 ############################################################################
-sub any_arg_is {
-  my $match = shift;
-  foreach my $arg (@ARGV) {
-    if ($arg eq $match) {
-      return 1;
-    }
-  }
-  return 0;
-}
-
-sub numeric_arg_value {
-  my $name = shift;
-  my @args = @ARGV[1..$#ARGV];
-  my $arg_str = join(" ", @args);
-  if ($arg_str =~ /$name ?=? ?([0-9]+)/) {
-    return $1;
-  }
-}
-
-sub string_arg_value {
-  my $name = shift;
-  my @args = @ARGV[1..$#ARGV];
-  my $arg_str = join(" ", @args);
-  if ($arg_str =~ /$name ?=? ?([^ ]+)/) {
-    return $1;
-  }
-}
 
 my $status = 0;
 
-my $print_help = any_arg_is("-h") || any_arg_is("--help");
-my $print_list = any_arg_is("--list");
-my $is_micro = any_arg_is("--micro");
-my $github_user = string_arg_value("--github-user") || $default_github_user;
-my $branch = string_arg_value("branch");
-
-if ($is_micro && !$branch) {
-  die "For micro releases, you must define the branch you want to use with --branch";
-}
-
-# Process Workspace Argument
+# Process Optional Arguments
+my $print_help = 0;
+my $print_list = 0;
+my $print_list_all = 0;
+my $step_expr = 0;
+my $remedy = 0;
+my $force = 0;
+my $remote = $default_remote;
+my $branch = $default_branch;
+my $github_user = $default_github_user;
+my $download_url = $default_download_url;
+my $micro = 0;
+my $next_version = "";
+my $version_id = $default_post_release_version_id;
+my $skip_devguide = 0;
+my $skip_doxygen = 0;
+my $skip_website = 0;
+my $skip_ftp = 0;
+
+GetOptions(
+  'help!' => \$print_help,
+  'list!' => \$print_list,
+  'list-all!' => \$print_list_all,
+  'steps=s' => \$step_expr,
+  'remedy!' => \$remedy,
+  'force!' => \$force,
+  'remote=s' => \$remote,
+  'branch=s' => \$branch,
+  'github-user=s' => \$github_user,
+  'download-url=s' => \$download_url,
+  'micro!' => \$micro,
+  'next-version=s' => \$next_version,
+  'version-id=s' => \$version_id,
+  'skip-devguide!' => \$skip_devguide,
+  'skip-doxygen!' => \$skip_doxygen ,
+  'skip-website!' => \$skip_website ,
+  'skip-ftp!' => \$skip_ftp,
+) or die "See --help for options.\nStopped";
+
+if (!(scalar(@ARGV) == 0 || scalar(@ARGV) == 2)) {
+  die "Expecting at 0 or 2 positional arugments. See --help.\nStopped";
```

Comment:
```suggestion
  die "Expecting 0 or 2 positional arguments. See --help.\nStopped";
```

Reply:
For some reason the spell checker doesn't work in perl strings in my vim. It's kinda annoying. For the help message I piped it into vim to spell check it.